src/resty/aws/credentials/ChainableTemporaryCredentials.lua

```diff
@@ -104,21 +104,30 @@ end
 -- updates credentials.
 -- @return success, or nil+err
 function ChainedTemporaryCredentials:refresh()
-  local result, err = self.sts:assumeRole(self.params)
-  if not result then
-    return nil, err
+  local response, err = self.sts:assumeRole(self.params)
+  if not response then
+    return nil, "Request for token data failed: " .. tostring(err)
   end
 
-  if type(result) == "string" then
-    local res, err = cjson.decode(result)
-    if not res then
-      return nil, "failed to json-decode assumeRole results with '" .. err .. "', input: " .. result
-    end
-    result = res
+  if response.status ~= 200 then
+    return nil, ("request for token returned '%s': %s"):format(tostring(response.status), response.body)
+  end
+
+  local resp_body_lom, err = lom.parse(response.body)
```

Comment:
the old code did a check on the body being a string, is it safe to drop that check?

Reply:
I added it back by checking if `response.body` is a string